megamek/src/megamek/utilities/ai/BehaviorParameters.java

```diff
@@ -0,0 +1,185 @@
+            rand.nextGaussian(), rand.nextGaussian(), rand.nextGaussian(),
+            rand.nextGaussian(), rand.nextGaussian(), rand.nextGaussian(),
+            rand.nextGaussian(), rand.nextGaussian(), rand.nextGaussian(),
+            rand.nextGaussian(), rand.nextGaussian(), rand.nextGaussian(),
+            rand.nextGaussian(), rand.nextGaussian(), rand.nextGaussian(),
+            rand.nextGaussian(), rand.nextGaussian(), rand.nextGaussian(),
+            rand.nextGaussian() * 1.5, rand.nextGaussian() * 1.5,
+            rand.nextGaussian() * 1.5, rand.nextGaussian() * 1.5,
+            rand.nextGaussian() * 1.5, rand.nextGaussian() * 1.5,
+            rand.nextGaussian() * 1.5, rand.nextGaussian() * 1.5);
+        return behaviorParameters.clamp(0, 1);
+    }
+
+    public BehaviorParameters multiply(double scalar) {
+        return new BehaviorParameters(
+            p1 * scalar, p2 * scalar, p3 * scalar,
+            p4 * scalar, p5 * scalar, p6 * scalar,
+            p7 * scalar, p8 * scalar, p9 * scalar,
+            p10 * scalar, p11 * scalar, p12 * scalar,
+            p13 * scalar, p14 * scalar, p15 * scalar,
+            p16 * scalar, p17 * scalar, p18 * scalar,
+            p19 * scalar, p20 * scalar, p21 * scalar,
+            p22 * scalar, p23 * scalar, p24 * scalar,
+            p25 * scalar, p26 * scalar, p27 * scalar,
+            p28 * scalar, p29 * scalar
+        );
+    }
+
+    public BehaviorParameters clamp(double min, double max) {
+        return new BehaviorParameters(
+            clamp(p1, min, max), clamp(p2, min, max), clamp(p3, min, max),
+            clamp(p4, min, max), clamp(p5, min, max), clamp(p6, min, max),
+            clamp(p7, min, max), clamp(p8, min, max), clamp(p9, min, max),
+            clamp(p10, min, max), clamp(p11, min, max), clamp(p12, min, max),
+            clamp(p13, min, max), clamp(p14, min, max), clamp(p15, min, max),
+            clamp(p16, min, max), clamp(p17, min, max), clamp(p18, min, max),
+            clamp(p19, min, max), clamp(p20, min, max), clamp(p21, min, max),
+            clamp(p22, min, max*1.5), clamp(p23, min, max*1.5), clamp(p24, min, max*1.5),
+            clamp(p25, min, max*1.5), clamp(p26, min, max*1.5), clamp(p27, min, max*1.5),
+            clamp(p28, min, max*1.5), clamp(p29, min, max*1.5)
+        );
+    }
+
+    public double clamp(double value, double min, double max) {
+        return Math.max(min, Math.min(max, value));
+    }
+
+    public BehaviorParameters perturb(int index, double epsilon) {
+        double[] params = toArray();
+        params[index] += epsilon;
+        return fromArray(params);
+    }
+
+    public BehaviorParameters set(int index, double value) {
+        double[] params = toArray();
+        params[index] = value;
+        return fromArray(params);
+    }
+
+    public double get(int index) {
+        double[] params = toArray();
+        return params[index];
+    }
+
+    public int size() {
+        return 28;
```

Comment:
Good lord, no :)

Reply:
That explains alot auehhuaehueahueahueauh